### PR TITLE
Integrate latest ledger, consensus, api and cli for 8.6.0 

### PR DESCRIPTION
**Pull request by teodanciu**
# Description

Integration with latest ledger, consensus,  cardano-api  8.28.0.0- based. 

# Checklist

- [ ] Commit sequence broadly makes sense and commits have useful messages
- [ ] New tests are added if needed and existing tests are updated.  These may include:
  - golden tests
  - property tests
  - roundtrip tests
  - integration tests
  See [Runnings tests](https://github.com/input-output-hk/cardano-node-wiki/wiki/Running-tests) for more details
- [ ] Any changes are noted in the `CHANGELOG.md` for affected package
- [ ] The version bounds in `.cabal` files are updated
- [ ] CI passes. See note on CI.  The following CI checks are required:
  - [ ] Code is linted with `hlint`.  See `.github/workflows/check-hlint.yml` to get the `hlint` version
  - [ ] Code is formatted with `stylish-haskell`.  See `.github/workflows/stylish-haskell.yml` to get the `stylish-haskell` version
  - [ ] Code builds on Linux, MacOS and Windows for `ghc-8.10.7` and `ghc-9.2.7`
- [ ] Self-reviewed the diff

# Note on CI
If your PR is from a fork, the necessary CI jobs won't trigger automatically for security reasons.
You will need to get someone with write privileges.  Please contact IOG node developers to do this
for you.

### CODE REVIEW

**Review by carbolymer — APPROVED** [review]
LGTM. I'm not 100% sure about tracing changes but they look sensible to me.

**Review by Jimbo4350 — APPROVED** [review]
LGTM, I would talk to benchmarking regarding the tracers.

**Review by Jimbo4350 — APPROVED** [review]
Looked at again, LGTM! Thank you @teodanciu  :)

File: cardano-node/src/Cardano/Node/Tracing/Era/Shelley.hs
Comment (jutaro) [inline cardano-node/src/Cardano/Node/Tracing/Era/Shelley.hs:168]: Has the underlying datatype removed this constructor?  And does InvalidPrevGovActionIdsInProposals replace it?

**Review by jutaro — CHANGES_REQUESTED** [review]
Just a request for one little change. Otherwise it looks good to me

File: cardano-node/src/Cardano/Node/Tracing/Era/HardFork.hs
Comment (teodanciu) [reply cardano-node/src/Cardano/Node/Tracing/Era/HardFork.hs:356]: Not sure I understand, do you mean something like this: 

```
forMachine = const $ forMachine DMinimal
``` 
?

**Comment by teodanciu** [comment]
@jutaro @carbolymer Could you have a look at my last commit? I'm uncertain that [my interpretation](https://github.com/input-output-hk/cardano-node/pull/5516#discussion_r1375270491) of this change request is correct.

File: cardano-node/src/Cardano/Node/Tracing/Era/HardFork.hs
Comment (carbolymer) [reply cardano-node/src/Cardano/Node/Tracing/Era/HardFork.hs]: I'm not sure. It would be the best if @jutaro could confirm that.

Comment (amesgen) [reply cardano-node/src/Cardano/Node/Tracing/Era/HardFork.hs]: I think
```haskell
forMachine = const $ forMachine DMinimal
```
will just loop at runtime as it is calling `forMachine` instantiated at the same type, or am I missing sth?

**Review by NadiaYvette — APPROVED** [review]
The tx-generator changes look like straightforward adaptations to the API changes. The tracing change also makes sense from what I can get by reading the code.

File: cardano-node/src/Cardano/Node/Tracing/Era/HardFork.hs
Comment (teodanciu) [reply cardano-node/src/Cardano/Node/Tracing/Era/HardFork.hs]: Sounds like it's safer to remove this commit and leave the original version (which presumably has been tested),  and then improve the code after the release, instead of risking this blowing up at runtime (unless someone knowledgeable of this can  clarify). 
On the other hand, merging of the PR (and thus of the release) is blocked until the requested changes have been approved.

Comment (jutaro) [reply cardano-node/src/Cardano/Node/Tracing/Era/HardFork.hs]: I was thinking like: `forMachine dtal = forMachine DMinimal . getHardForkSelectView`, but then you have to look that in the concrete instances of the different selectViews, and implement the forMachine code, so that it doesn't print the blockNr., if the detail level is minimal.

Comment (teodanciu) [reply cardano-node/src/Cardano/Node/Tracing/Era/HardFork.hs:359]: I removed the commit and added a TODO with the explanation from this thread.  Resolving the conversation.

**Pull request merged**
